src/ralph/data_center/models/physical.py

```diff
@@ -52,6 +53,30 @@
 ]
 
 
+def assign_additional_hostname_choices(actions, objects):
+    """
+    Generate choices with networks common for each object.
+
+    Args:
+        actions: Transition action list
+        objects: Django models objects
+
+    Returns:
+        list of tuples with available network choices
+    """
+    network_environments = []
+    networks = []
+    for obj in objects:
+        network_environments.append(
+            set(obj._get_available_network_environments())
+        )
+    network_environments = set.intersection(*network_environments)
+    for environment in network_environments:
+        for network in environment.network_set.all():
```

Comment:
shouldn't we allow only for networks which are available for all considered objects (not all networks in these networks environments)? I would not involve network environment here and focus only on networks - consider using `obj. _get_available_networks`